src/hooks/useRequiresAccount.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  showRequiresAccountModal,
+  updateRouteOnExit
+} from 'containers/sign-on/store/actions'
+import { useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { getAccountUser } from 'store/account/selectors'
+import { useSelector } from 'utils/reducer'
+import { push as pushRoute } from 'connected-react-router'
+import { SIGN_UP_PAGE } from 'utils/route'
+
+/**
+ * Checks that a user is signed in, else redirects to the sign in
+ * page with a banner saying - you need an account to do that
+ */
+export const useRequiresAccount = (route?: string) => {
+  const account = useSelector(getAccountUser)
+  const dispatch = useDispatch()
+  useEffect(() => {
+    if (!account) {
+      if (route) dispatch(updateRouteOnExit(route))
```

Comment:
should this be an if / else?

Reply:
This should remain an if statement. This just updates the route to go to when exiting the signup page. 
This function mimics the sagaHelper. https://github.com/AudiusProject/audius-client/blob/a175e0f87d962f7e1d07483339bf6caad1171cb0/src/utils/sagaHelpers.js#L111-L121